ringvax/app.py

```diff
@@ -363,14 +363,42 @@ def infectiousness_callback():
             )
 
             st.header("Number of infections")
-            st.write(
-                f"Distribution of the total number of infections seen in {n_generations} generations."
-            )
+            with st.expander("Plot Options"):
+                generational_counts = get_generational_infection_count_df(sim_df)
+                plot_gen = st.segmented_control(
+                    "Generation to plot",
+                    options=range(1, n_generations + 1),
+                    default=n_generations,
+                )
+                cumulative = (
+                    True
+                    if st.segmented_control(
+                        "Show infections cumulatively or in specific generation?",
+                        options=["Cumulative", "In generation"],
+                        default="Cumulative",
+                    )
+                    == "Cumulative"
+                    else False
+                )
+
+                if cumulative:
+                    counts = (
+                        generational_counts.filter(pl.col("generation") <= plot_gen)
+                        .group_by("simulation")
+                        .agg(pl.col("count").sum())
+                    )
+                else:
+                    counts = generational_counts.filter(
+                        pl.col("generation") == plot_gen
+                    )
+
+            size_hist_title = f"Distribution of the {'cumulative ' if cumulative else ''}number of infections seen {'up to and including' if cumulative else 'in'} generation {plot_gen}."
+            st.write(size_hist_title)
             st.altair_chart(
-                alt.Chart(get_total_infection_count_df(sim_df))
+                alt.Chart(counts)
                 .mark_bar()
                 .encode(
-                    x=alt.X("size:Q", bin=True, title="Number of infections"),
+                    x=alt.X("count:Q", bin=True, title="Number of infections"),
                     y=alt.Y("count()", title="Count"),
```

Comment:
```suggestion
                    y=alt.Y("count()", title="Number of simulations"),
```
I think this is what's going on? I got confused because now we have "count" on both axes

Reply:
Well, they are both counts. But yes, the Y axis is counts of things measured in simulations, and the X axis is a count of infections in those simulations. And the dataframe column "count" is "count of infections in generations" which ends up on the X axis. Maybe I should rename that too